src/fragmentarium/ui/search/TransliterationSearchForm.tsx

```diff
@@ -37,41 +35,19 @@ type State = {
   transliteration: string
 }
 type Props = {
-  transliteration: string | null | undefined
-} & RouteComponentProps
+  onChange(searchForm: string, searchQuery: string): void
+  getState(key: string): string
+}
 
 class TransliterationSearchForm extends Component<Props, State> {
-  state = {
-    transliteration: this.props.transliteration || '',
-  }
-
   onChange = (event) => {
-    this.setState({
-      transliteration: event.target.value,
-    })
-  }
-
-  submit = (event) => {
-    event.preventDefault()
-    this.props.history.push(
-      `/fragmentarium/search/?${stringify({
-        transliteration: this.state.transliteration,
-      })}`
-    )
-  }
-
-  componentDidUpdate(prevProps, prevState, snapshot) {
-    if (this.props.transliteration !== prevProps.transliteration) {
-      this.setState({
-        transliteration: this.props.transliteration || '',
-      })
-    }
+    this.props.onChange('transliteration', event.target.value || '')
```

Comment:
You should only pass the value and not couple the implementation with something on the outside. Whoever defines the callback should make do the correct thing without the form knowing its internals.